python/pylibcudf/pylibcudf/strings/findall.pyx

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+from libcpp.memory cimport unique_ptr
+from libcpp.utility cimport move
+from pylibcudf.column cimport Column
+from pylibcudf.libcudf.column.column cimport column
+from pylibcudf.libcudf.strings cimport findall as cpp_findall
+from pylibcudf.strings.regex_program cimport RegexProgram
+
+
+cpdef Column findall(Column input, RegexProgram prog):
+    """
+    Returns a lists column of strings for each matching occurrence using
+    the regex_program pattern within each string.
```

Comment:
Can we call this argument `pattern` (rather than `prog`). It's kind of a leaky implementation detail that the type is called `Program`?